nexus/src/external_api/console_api.rs

```diff
@@ -552,75 +272,6 @@ pub(crate) async fn login_begin(
         .await
 }
 
-pub(crate) async fn console_index_or_login_redirect(
-    rqctx: RequestContext<ApiContext>,
-) -> Result<Response<Body>, HttpError> {
-    let opctx = crate::context::op_context_for_external_api(&rqctx).await;
-
-    // if authed, serve console index.html with JS bundle in script tag
-    if let Ok(opctx) = opctx {
-        if opctx.authn.actor().is_some() {
-            return serve_console_index(rqctx).await;
-        }
-    }
-
-    // otherwise redirect to idp
-
-    // Put the current URI in the query string to redirect back to after login.
-    let redirect_uri = rqctx
-        .request
-        .uri()
-        .path_and_query()
-        .map(|p| p.to_string().parse::<RelativeUri>())
-        .transpose()
-        .map_err(|e| {
-            HttpError::for_internal_error(format!("parsing URI: {}", e))
-        })?;
-    let login_url = get_login_url(&rqctx, redirect_uri).await?;
-
-    Ok(Response::builder()
-        .status(StatusCode::FOUND)
-        .header(http::header::LOCATION, login_url)
-        .body("".into())?)
-}
-
-// Dropshot does not have route match ranking and does not allow overlapping
-// route definitions, so we cannot use a catchall `/*` route for console pages
-// because it would overlap with the API routes definitions. So instead we have
-// to manually define more specific routes.
-
-macro_rules! console_page {
-    ($name:ident) => {
-        pub(crate) async fn $name(
-            rqctx: RequestContext<ApiContext>,
-        ) -> Result<Response<Body>, HttpError> {
-            console_index_or_login_redirect(rqctx).await
-        }
-    };
-}
-
-// only difference is the _path_params arg
-macro_rules! console_page_wildcard {
-    ($name:ident) => {
-        pub(crate) async fn $name(
-            rqctx: RequestContext<ApiContext>,
-            _path_params: Path<params::RestPathParam>,
-        ) -> Result<Response<Body>, HttpError> {
-            console_index_or_login_redirect(rqctx).await
-        }
-    };
-}
-
-console_page_wildcard!(console_projects);
-console_page_wildcard!(console_settings_page);
-console_page_wildcard!(console_system_page);
-console_page_wildcard!(console_lookup);
-console_page!(console_root);
-console_page!(console_projects_new);
-console_page!(console_silo_images);
-console_page!(console_silo_utilization);
-console_page!(console_silo_access);
-
```

Comment:
I created these to reduce boilerplate back when the actual endpoint handlers were defined here (rather that silly little helper functions called by the endpoint handlers, which is what they are now). Because of the change to Dropshot API, we kinda have to have the boilerplate over in `http_entrypoints.rs` anyway, so these are pointless.